lib/parsers/ass_to_json.js

```diff
@@ -0,0 +1,40 @@
+const { trim, zipObj, reduce } = require('ramda');
+
+const dialogueKeysRegex = /\[Events\]\nFormat:(.*)\n/;
+const dialogueLineRegex = /Dialogue:/;
+
+const getKeys = (subtitleText) => {
+  const [, keysLine] = subtitleText.match(dialogueKeysRegex);
+  const keys = keysLine.split(',').map(trim);
+  return keys;
+};
+
+
+const isDialogueLine = (line) => dialogueLineRegex.test(line);
+
+const ASStoJSON = (subtitleText) => {
+  // split file by line
+  const lines = subtitleText.split(/\r\n|\n|\r/);
+
+  // get the keys of the dialogue lines
+  const keys = getKeys(subtitleText);
+
+  const zipDialogueReducer = (acc, line) => {
+    if (!isDialogueLine(line)) return acc;
+
+    // eslint-disable-next-line no-useless-escape
+    const fixedNewlines = line.replace('\N', '\n');
```

Comment:
What's the purpose of this? Should `\N` be included on line 17 to split on?

Reply:
line 17 is splitting the whole file by each line in order to target each timecode entry, whereas `\N` is used to indicate a break in text for the same entry.
For example, one line in the ASS file would look like this:
```
Dialogue: 0,0:03:18.77,0:03:24.77,Default,,0000,0000,0000,,Line of Text 1 \NLine of Text 2.
```
Splitting by `\N` would treat `Line of Text 2` as a new line in the file and would break how `isDialogueLine` is evaluated.